.github/labels.yml

```diff
@@ -47,46 +47,46 @@
   aliases: []
   description: issue requires detective debug work to figure out what's going wrong.
 
-- name: A-design
-  color: "#FBCA04"
-  aliases: []
-  description: issue requires design work to think about how it would best be accomplished.
-
 # Describes the type of issue or pull request.
 - name: T-bug
   color: "#FEF2C0"
   aliases: []
   description: this issue covers unexpected and/or wrong behaviour.
 
-- name: T-feat
+- name: T-implementation
   color: "#FEF2C0"
   aliases: []
   description: this issue/pr is a new feature or functionality.
 
-- name: T-enhancement
+- name: T-research
   color: "#FEF2C0"
   aliases: []
-  description: this issue/pr covers improvement of existing functionality.
+  description: this issue/pr is a research type issue.
 
-- name: T-refactor
+- name: T-investigation
   color: "#FEF2C0"
   aliases: []
-  description: this issue/pr covers refactoring of existing code.
+  description: this issue/pr is an investigation, probably related to some bug with unknown causes.
 
-- name: T-security
+- name: T-design
+  color: "#FEF2C0"
+  aliases: [ ]
+  description: this issues describes design requirements
+
+- name: T-enhancement
   color: "#FEF2C0"
   aliases: []
-  description: this issue/pr covers security sensitive problem.
+  description: this issue/pr covers improvement of existing functionality.
 
-- name: T-research
+- name: T-refactor
```

Comment:
Can we just get rid of this one? Most of what is called refactoring is actually an enhancement of some kind, like improving an API or making some code accessible for multiple consumers.

Refactoring was supposed to mean changing the code without changing the behavior (including the API). If the unit tests need to be changed, it's not refactoring. I'm not an agile guru and everyone uses refactoring much more broadly, but regardless of that I'm skeptical that it is useful as an issue type or label.

Reply:
I would then rather remove enhancement since refactoring is exactly what you describe. wdyt?

Reply:
I'm fine with removing enhancement instead. We only have a few issues tagged with either of them so should be quick to move them all to one or the other.